pkg/netceptor/ping.go

```diff
@@ -75,20 +88,30 @@ func (s *Netceptor) Ping(ctx context.Context, target string, hopsToLive byte) (t
 		return time.Since(startTime), "", fmt.Errorf("timeout")
 	case <-ctxPing.Done():
 		return time.Since(startTime), "", fmt.Errorf("user cancelled")
-	case <-s.context.Done():
+	case <-s.Context().Done():
 		return time.Since(startTime), "", fmt.Errorf("netceptor shutdown")
 	}
 }
 
+type NetcForTraceroute interface {
+	MaxForwardingHops() byte
+	Ping(ctx context.Context, target string, hopsToLive byte) (time.Duration, string, error)
+	Context() context.Context
+}
+
 // TracerouteResult is the result of one hop of a traceroute.
 type TracerouteResult struct {
 	From string
 	Time time.Duration
 	Err  error
 }
 
-// Traceroute returns a channel which will receive a series of hops between this node and the target.
 func (s *Netceptor) Traceroute(ctx context.Context, target string) <-chan *TracerouteResult {
+	return CreateTraceroute(ctx, s, target)
+}
+
+// Traceroute returns a channel which will receive a series of hops between this node and the target.
```

Comment:
Please update the comments :) 

I like the names you picked here as a `Traceroute` `chan` is returned, so it makes sense